src/main/java/lc/kra/system/keyboard/GlobalKeyboardHook.java

```diff
@@ -132,19 +138,66 @@ public GlobalKeyboardHook(boolean raw) throws UnsatisfiedLinkError {
 	 * @param event A global key event
 	 */
 	private void keyPressed(GlobalKeyEvent event) {
+		
+		int virtualKeyCode = event.getVirtualKeyCode();
+		
+		// only add if key is not already hold down
+		if(!isKeyHoldDown(event))
```

Comment:
This check is unnecessary, you are already using as Set, which means directly adding the virtualKeyCode is just fine. The Set will perform a (more efficient) check anyways when you call "add".

Reply:
Haha I was actively thinking "this must be a set so it can only be in there once" and still go ahead with checking for it... :D I guess i was tired

Done